File: generate_icons.py
from PIL import Image, ImageDraw, ImageFont
import os

def generate_icon(size):
    # Create a new image with transparent background
    image = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(image)
    
    # Calculate font size (70% of icon size)
    font_size = int(size * 0.7)
    
    # Try to use Arial, fall back to default if not available
    try:
        font = ImageFont.truetype("Arial", font_size)
    except:
        font = ImageFont.load_default()
    
    # Draw the letter G in black
    text = "G"
    text_bbox = draw.textbbox((0, 0), text, font=font)
    text_width = text_bbox[2] - text_bbox[0]
    text_height = text_bbox[3] - text_bbox[1]
    
    # Center the text
    x = (size - text_width) // 2
    y = (size - text_height) // 2
    
    # Draw the text
    draw.text((x, y), text, font=font, fill='black')
    
    return image

# Create images directory if it doesn't exist
if not os.path.exists('images'):
    os.makedirs('images')

# Generate icons for all sizes
sizes = [16, 48, 128]
for size in sizes:
    icon = generate_icon(size)
    icon.save(f'images/icon{size}.png')
    print(f'Generated icon{size}.png')

File: generate_icons.js
const { createCanvas } = require('canvas');
const fs = require('fs');
const path = require('path');

function generateIcon(size) {
    const canvas = createCanvas(size, size);
    const ctx = canvas.getContext('2d');
    
    // Clear background (transparent)
    ctx.clearRect(0, 0, size, size);
    
    // Set text properties
    ctx.fillStyle = '#000000';
    ctx.textAlign = 'center';
    ctx.textBaseline = 'middle';
    
    // Adjust font size based on icon size
    const fontSize = Math.floor(size * 0.7);
    ctx.font = `bold ${fontSize}px Arial`;
    
    // Draw the letter G
    ctx.fillText('G', size/2, size/2);
    
    return canvas;
}

// Ensure images directory exists
const imagesDir = path.join(__dirname, 'images');
if (!fs.existsSync(imagesDir)) {
    fs.mkdirSync(imagesDir);
}

// Generate icons for all sizes
const sizes = [16, 48, 128];

sizes.forEach(size => {
    const canvas = generateIcon(size);
    const buffer = canvas.toBuffer('image/png');
    fs.writeFileSync(path.join(imagesDir, `icon${size}.png`), buffer);
    console.log(`Generated icon${size}.png`);
});
